src/goodHealthApp/models.py

```diff
@@ -1,16 +1,40 @@
 from django.contrib.auth.models import User
 from django.db import models
+from django.conf import settings
 from django.utils.timezone import now
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+
+def get_upload_image_dir(instance, filename):
+    return "uploads/{mimeType}/{userid}/{file}".format(mimeType="images", userid=instance.user.username, file=filename)
```

Comment:
I think the depth of this path is not needed considering each user can only have one avatar at any point in that. consider using a single folder for the avatars where each avatar bears the user's username as the filename. something like "uploads/avatars/elizabethadegbaju.png" rather than "uploads/images/elizabethadegbaju/filenameasuploaded.png"